core/src/cluster_info.rs

```diff
@@ -2462,7 +2462,7 @@ impl ClusterInfo {
         while let Ok(packet) = requests_receiver.try_recv() {
             let num_packets = packets.len() + packet.packets.len();
             if num_packets > MAX_GOSSIP_TRAFFIC {
-                let excess_count = usize::min(MAX_GOSSIP_TRAFFIC - num_packets, packets.len());
+                let excess_count = usize::min(num_packets - MAX_GOSSIP_TRAFFIC, packets.len());
```

Comment:
Is there a reason to take the `min` here?

For instance if on the last iteration of this loop, num_packets = 400, MAX_GOSSIP_TRAFFIC = 100 and packets.len() = 1,  then we'll drop at most 1 packet and end up with `4x` the MAX_GOSSIP_TRAFFIC

Reply:
I updated the code to enforce the max-traffic more strictly. but to answer your question,  we cannot call `packets.drain` with range exceeding `packets.len`. So we needed a `min` there. I changed the code to call `drain` after `extend`, and so no need for `min` anymore. However the previous code was more similar to [current mainnet code](https://github.com/solana-labs/solana/blob/8838a55a1/core/src/cluster_info.rs#L2490-L2497), which allows the length overrun the max-gossip-traffic.
Either way I do not think it does matter much here, because the overshoot will be likely very small relatively.

Reply:
I see, thanks!